### ISSUE TITLE
bundle not finding compatible versions for several gems

### ISSUE THREAD

**Issue by jtbarker** [bundler/bundler#7034]
bundle install results in this:

```
➜  collabfield git:(master) ✗ bundle install
The dependency tzinfo-data (>= 0) will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x86-mswin32, x64-mingw32, java. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x86-mswin32 x64-mingw32 java`.
Fetching gem metadata from https://rubygems.org/............
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Bundler could not find compatible versions for gem "activesupport":
  In snapshot (Gemfile.lock):
    activesupport (= 4.2.11.1)

  In Gemfile:
    rails (~> 5.1.0) was resolved to 5.1.2.rc1, which depends on
      activesupport (= 5.1.2.rc1)

    railties was resolved to 4.2.11.1, which depends on
      activesupport (= 4.2.11.1)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```

bundle env results:

https://gist.github.com/jtbarker/563a032bb380a0f7266a5c6887a0daef

**Comment by deivid-rodriguez** [comment]
Hello @jtbarker.

This is a problem with your Gemfile, not with bundler. We'll keep working on better error messages though.

In this case, bundler is reporting a conflict because your lock file is locked to `rails 4.2.11.1`, but your `rails` requirement is `rails ~> 5.1.0`. Those do not match, because `rails ~> 5.1.0` means `rails >= 5.1.0, < 5.2`, and `4.2.11.1` does not satisfy that.

Please, try the suggestion `bundler` is giving you (running `bundle update`) and carefully read through the error messages to find conflicting dependencies/requirements and fix them. :+1:

Closing!

**Comment by deivid-rodriguez** [comment]
@jtbarker I just saw the `bundle update` error you posted on Slack. In that case, also dependency incompatibility:

```
Bundler could not find compatible versions for gem "actionpack":
  In Gemfile:
    actionpack
    rails (~> 5.1.0) was resolved to 5.1.2.rc1, which depends on
      actionpack (= 5.1.2.rc1)
Bundler could not find compatible versions for gem "activesupport":
  In Gemfile:
    activesupport
    rails (~> 5.1.0) was resolved to 5.1.2.rc1, which depends on
      activesupport (= 5.1.2.rc1)
    rspec-rails (= 3.3.2) was resolved to 3.3.2, which depends on
      activesupport (>= 3.0, < 4.3)
Bundler could not find compatible versions for gem "railties":
  In Gemfile:
    railties
    rails (~> 5.1.0) was resolved to 5.1.2.rc1, which depends on
      railties (= 5.1.2.rc1)
```

Your `Gemfile` includes `gem "rspec-rails", "3.3.2"`, which depends on `activesupport >= 3.0, < 4.3`, and also `gem "rails",  "~> 5.1.0"`, which depends on a version of `activesupport` incompatible with the requirement of `rspec-rails`.

Relax your `rspec-rails` requirement and you should be alright. For example,

```diff
- gem 'rspec-rails', '3.3.2'
+ gem 'rspec-rails', '~> 3.3'
```
